src/transport/SessionHandle.h

```diff
@@ -21,31 +21,51 @@ namespace chip {
 
 class SecureSessionMgr;
 
-class SecureSessionHandle
+class SessionHandle
 {
 public:
-    SecureSessionHandle() : mPeerNodeId(kPlaceholderNodeId), mPeerKeyId(0), mFabric(Transport::kUndefinedFabricIndex) {}
-    SecureSessionHandle(NodeId peerNodeId, uint16_t peerKeyId, FabricIndex fabric) :
-        mPeerNodeId(peerNodeId), mPeerKeyId(peerKeyId), mFabric(fabric)
-    {}
+    SessionHandle() : mPeerNodeId(kPlaceholderNodeId), mFabric(Transport::kUndefinedFabricIndex) {}
+
+    SessionHandle(NodeId peerNodeId, FabricIndex fabric) : mPeerNodeId(peerNodeId), mFabric(fabric) {}
+
+    SessionHandle(NodeId peerNodeId, uint16_t localKeyId, uint16_t peerKeyId, FabricIndex fabric) :
+        mPeerNodeId(peerNodeId), mFabric(fabric)
+    {
+        mLocalKeyId.SetValue(localKeyId);
+        mPeerKeyId.SetValue(peerKeyId);
+    }
 
     bool HasFabricIndex() const { return (mFabric != Transport::kUndefinedFabricIndex); }
     FabricIndex GetFabricIndex() const { return mFabric; }
     void SetFabricIndex(FabricIndex fabricId) { mFabric = fabricId; }
 
-    bool operator==(const SecureSessionHandle & that) const
+    bool operator==(const SessionHandle & that) const { return mPeerNodeId == that.mPeerNodeId && mPeerKeyId == that.mPeerKeyId; }
+
+    bool match(const SessionHandle & that) const
```

Comment:
foo.match(bar) is not obvious ... match what?

How about 'compatibleLocalKey' or similar to make the code easier to read?

Reply:
This method is introduced for match exchange from session, how about matchExchange?

Reply:
Rename to MatchIncomingSession for now, please lets me know if you have a better name